scrapinghub/client/items.py

```diff
@@ -59,3 +73,30 @@ def _modify_iter_params(self, params):
         if offset:
             params['start'] = '{}/{}'.format(self.key, offset)
         return params
+
+    def list_iter(self, chunksize=10000, *args, **kwargs):
+        """An alternative interface for reading items by returning them
+        as a generator which yields lists of items sized as `chunksize`.
+
+        This is a convenient method for cases when processing a large amount of
+        items from a job isn't ideal in one go due to the large memory needed.
+        Instead, this allows you to process it chunk by chunk.
+
+        You can improve I/O overheads by increasing the chunk value but that
+        would also increase the memory consumption.
+
+        :return: an iterator over items, yielding lists of items.
+        :rtype: :class:`collections.Iterable`
+        """
+
+        processed = 0
+        while True:
+            next_key = self.key + '/' + str(processed)
+            items = [
+                item for item in self.iter(
+                    count=chunksize, start=next_key, *args, **kwargs)
+            ]
+            yield items
+            processed += len(items)
+            if len(items) < chunksize:
+                break
```

Comment:
I haven't considered this use case since we pretty much consume the entire items in a job. Much appreciated @manycoding! I've added some docstrings and test case as well.

Though the CI tests will fail because of the current project setup on `vcr-py`. @vshlapakov is it okay to mock this at the moment or should we standardize it to use cassettes like the current tests?